packages/components/src/components/icon/icon.component.ts

```diff
@@ -126,15 +127,18 @@ class Icon extends Component {
    */
   private async getIconData() {
     if (this.iconProviderContext.value) {
-      const { fileExtension, url, iconsCache, shouldCache } = this.iconProviderContext.value;
+      const { fileExtension, url, cacheName, cacheStrategy } = this.iconProviderContext.value;
       if (url && fileExtension && this.name) {
         // abort the previous fetch request if it is still pending
         // before retreiving from cache
         this.abortController.abort();
 
+        const iconsCache = cacheName && cacheStrategy === 'js-cache' ? jsIconsCache(cacheName) : undefined;
+
+        const iconData = iconsCache?.get(this.name);
```

Comment:
you need to use `?.` for the method call as well otherwise JS will try to call undefined because 

`iconsCache?.get()` means `(iconsCache ? iconsCache.get : undefined)()`

```suggestion
        const iconData = iconsCache?.get?.(this.name);
```